nethack_server/src/db_pgsql.c

```diff
@@ -424,23 +447,39 @@ db_set_user_password(int uid, const char *password)
 long
 db_add_new_game(int uid, const char *filename, const char *role,
                 const char *race, const char *gend, const char *align, int mode,
-                const char *plname, const char *levdesc)
+                const char *plname, const char *levdesc, int hp, int hpmax,
+                int en, int enmax, int wi, int in, int st, int dx,
+                int co, int ch)
 {
     PGresult *res;
-    char uidstr[16], modestr[16];
+    char uidstr[16], modestr[16], hpstr[16], hpmaxstr[16], enstr[16],
+         enmaxstr[16], wistr[16], intstr[16], ststr[16], dxstr[16], 
+         costr[16], chstr[16];
 
-    const char *const params[] = { filename, role, race, gend,
-        align, modestr, uidstr, plname, levdesc
+    const char *const params[] = { filename, role, race, gend, align, modestr,
+        uidstr, plname, levdesc, hpstr, hpmaxstr, enstr, enmaxstr, wistr,
+        intstr, ststr, dxstr, costr, chstr
     };
-    const int paramFormats[] = { 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+    const int paramFormats[] = { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+                                 0, 0, 0 };
     const char *gameid_str;
     int gid;
 
     sprintf(uidstr, "%d", uid);
     sprintf(modestr, "%d", mode);
+    sprintf(hpstr, "%d", hp);
+    sprintf(hpmaxstr, "%d", hpmax);
+    sprintf(enstr, "%d", en);
+    sprintf(enmaxstr, "%d", enmax);
+    sprintf(wistr, "%d", wi);
+    sprintf(intstr, "%d", in);
+    sprintf(ststr, "%d", st);
+    sprintf(dxstr, "%d", dx);
+    sprintf(costr, "%d", co);
+    sprintf(chstr, "%d", ch);
 
     res =
-        PQexecParams(conn, SQL_add_game, 9, NULL, params, NULL, paramFormats,
+        PQexecParams(conn, SQL_add_game, 19, NULL, params, NULL, paramFormats,
```

Comment:
Instead of hard-coding the number of params, you should calculate it. In this case, you can do it with sizeof(params) / sizeof(const char *)